### PR TITLE
feat(admob, ios): add adaptive banner support

### PR DESCRIPTION
**Pull request by dylancom**
### Description

This pull request adds support for adaptive banners. https://github.com/invertase/react-native-firebase/discussions/4276

### Release Summary

Added adaptive banner support for iOS.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### CODE REVIEW

File: packages/admob/lib/index.d.ts [hunk 1]
```diff
@@ -237,6 +237,11 @@ export namespace FirebaseAdMobTypes {
      */
     SMART_BANNER: 'SMART_BANNER';
 
+    /**
+     * A (next generation) dynamically sized banner that is full-width and auto-height.
+     */
+    APAPTIVE_BANNER: 'APAPTIVE_BANNER';
```
Comment (mikehardy) [inline packages/admob/lib/index.d.ts:243]:
```suggestion
    ADAPTIVE_BANNER: 'ADAPTIVE_BANNER';
```

File: packages/admob/lib/index.d.ts [hunk 2]
```diff
@@ -237,6 +237,11 @@ export namespace FirebaseAdMobTypes {
      */
     SMART_BANNER: 'SMART_BANNER';
 
+    /**
+     * A (next generation) dynamically sized banner that is full-width and auto-height.
```
Comment (mikehardy) [inline packages/admob/lib/index.d.ts:241]: If there is only an iOS implementation - should mention it is iOS only?

```suggestion
     * A (next generation) dynamically sized banner that is full-width and auto-height.
     *
     * iOS only
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hey thanks for this! Looks like a typo and maybe could use a note for iOS only, but otherwise if you have tested this (specifically the CFGloat / size calculations) and it works, that works for me

**Comment by dylancom** [comment]
> Hey thanks for this! Looks like a typo and maybe could use a note for iOS only, but otherwise if you have tested this (specifically the CFGloat / size calculations) and it works, that works for me

I have tested it and it works. I will fix the typo and add an iOS note. In the future I will probably have time to also add Android support.

**Review by mikehardy — APPROVED** [review]
Looks great! Thanks!
Until release, you can go to the "patch packages" action and find a zip file with the patch you can apply locally (using the always amazing https://github.com/ds300/patch-package). I mention that because I just did a couple of very big releases and I want those to marinate for a bit before I do another one

**Pull request merged**